### ISSUE TITLE
[DFC Orders] List products to import on screen

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#12301]
A basic product import has been done:

*  #12274

But there are some outstanding tasks from https://github.com/openfoodfoundation/openfoodnetwork/issues/12139:

* The input field "Enterprise" should be labelled "Create products for enterprise:".
* List of products to import is displayed on the screen.
* Each product has an import button to import that product into OFN. Technically, the form can contain the whole JSON and post it do the OFN DFC API to do the work. But another controller action or reflex can do this as well.
* Already imported products have a disabled import button.

**Comment by RaggedStaff** [comment]
Couple of enhancements/thoughts:

1. Already imported products have an "Update" button.
2. We need an option to import all
3. I think a modal that allows checking of certain products might be more user-frienldy (as in check to include Product in chosen enterprise)...
4. ... but that implies the option to remove the Product by unchecking the box. 

I think this is additional scope, but I'd like to get an estimate, with these extra options (assuming the current 3 days doesn't stand), @mkllnk ?

**Comment by dacook** [comment]
Hi @RaggedStaff (ccc @mkllnk ), the main requirement of this issue is ready for testing, however I note a few other things mentioned which may or may not be important. Attempting to summarise:
- [ ] The input field "Enterprise" should be labelled "Create products for enterprise:"
- [x] Products can be reviewed before import
- [x] Choose which new products to create
- [x] Choose which existing products to update
- [ ] a modal that allows checking of certain products*

* I'm not sure exactly what you meant by this. At first I imagined the product preview modal (available on Admin > Products > (product row) > `⠇` > Preview. But I guess you'd want to preview the product _as it would be imported_, which might be a bit of effort.
Another thought I had would be to show a summary of changes in the table. For example using green and red like a diff, or ~~strikethrough~~:
```diff
- Tomato sauce
+ Red sauce
```

But it wouldn't be practical to show a column for every single change. Hmm but maybe we could show that in a modal, for each product.

Also there are improvements mentioned by Konrad. Do these need fixing? If so, I'd need some more details about what the behaviour should be..

- [ ] Imported variants of other unit type have variant name = original name
- [ ] Imported variants of unit type 'item' have unit '1piece' now.
- [ ] Imported variants have price 0. (when new or existing)

Can you please confirm the next steps after the current PR?

**Comment by dacook** [comment]
PS Just remembered also something that I discussed with Maikel:

- [ ] Add shortcut to select/deselect all
   (or add more options like select only new, select only update)

**Comment by mkllnk** [comment]
> Imported variants have price 0.

Is that still the case? I thought that's solved. The price can even be calculated from the wholesale product.

**Comment by mkllnk** [comment]
> Imported variants of unit type 'item' have unit '1piece' now.

Maybe the label isn't right? OFN talks about items when DFC talks about pieces. I'm not sure if it matters but the label _1piece_ doesn't seem right.

**Comment by mkllnk** [comment]
> Imported variants of other unit type have variant name = original name

I'm not sure where the problem is here.

**Comment by dacook** [comment]
One more thing I noticed while working on this (although not directly related):
-  When importing over an existing product, the variant name is updated, but product name is not.

**Comment by mkllnk** [comment]
> When importing over an existing product, the variant name is updated, but product name is not.

Yes, because you might be importing multiple variants and you don't want to override the product name which may have been set in the OFN UI.

And I'm currently working on the DFC variants. The import can then have a DFC product that represents the Spree product and we can copy the name from there. So nothing worry about here.

**Comment by RaggedStaff** [comment]
Hi @dacook 

Thanks for all your work on this. For reference - it's usually best to ping me on slack for a quick response.

👍🏻  for the select/deselect all toggle.

Yes - it would be really helpful to have some sort of preview of what's changed on a product, perhaps a list of fields that have changed would be a good place to start, rather than trying to list actual values  (e.g. Update <PROD_ID> - changes to <Price>, <Description>, <Stock_Limit>) ? I'm aware that will probably make the UI too busy/clunky but not sure how else to make it work - could we have the Product Import list expandable to display details ? 

@mkllnk - the Price = 0 issue seems resolved to me. 

I'm not sure I'm  following the issue around units.

**Comment by mkllnk** [comment]
> it would be really helpful to have some sort of preview of what's changed on a product

We can do that but it's scope creep. I created an issue, estimate 2 days:

* #13169
